src/cargo/core/compiler/custom_build.rs

```diff
@@ -859,7 +911,19 @@ impl BuildOutput {
                 "warning" => warnings.push(value.to_string()),
                 "rerun-if-changed" => rerun_if_changed.push(PathBuf::from(value)),
                 "rerun-if-env-changed" => rerun_if_env_changed.push(value.to_string()),
-                _ => metadata.push((key.to_string(), value.to_string())),
+                "metadata" => {
+                    let (key, value) = parse_directive(whence.as_str(), line, &value)?;
```

Comment:
`parse_directive` would be giving the wrong error here, `cargo::KEY=VALUE` instead of `cargo::metadata=KEY=VALUE`

Reply:
Added `parse_metadata` to parse it. I don't try to reuse `parse_directive` because it makes the `parse_directive` function more confusing.